core/storage/collection/gae_models.py

```diff
@@ -108,15 +104,69 @@ def get_export_policy(cls):
             'language_code': base_models.EXPORT_POLICY.NOT_APPLICABLE,
             'tags': base_models.EXPORT_POLICY.NOT_APPLICABLE,
             'schema_version': base_models.EXPORT_POLICY.NOT_APPLICABLE,
-            'collection_contents': base_models.EXPORT_POLICY.NOT_APPLICABLE,
-            'nodes': base_models.EXPORT_POLICY.NOT_APPLICABLE
+            'collection_contents': base_models.EXPORT_POLICY.NOT_APPLICABLE
         })
 
     @classmethod
     def get_collection_count(cls):
         """Returns the total number of collections."""
         return cls.get_all().count()
 
+    @staticmethod
+    def convert_to_valid_dict(model_dict):
+        """Replace invalid fields and values in the CollectionModel dict.
+
+        Some old CollectionModels can contain fields
+        and field values that are no longer supported and would cause
+        an exception when we try to reconstitute a CollectionModel from
+        them. We need to remove or replace these fields and values.
+
+        Args:
+            model_dict: dict. The content of the model. Some fields and field
+                values might no longer exist in the CollectionModel
+                schema.
+
+        Returns:
+            dict. The content of the model. Only valid fields and values are
+            present.
+        """
+
+        # The nodes field is moved to collection_contents dict. We
+        # need to move the values from nodes field to collection_contents dict
+        # and delete nodes.
+        if 'nodes' in model_dict and model_dict['nodes']:
+            if ('collection_contents' not in model_dict or not
+                    model_dict['collection_contents']
+               ):
+                model_dict['collection_contents'] = (
+                    (datastore_services.JsonProperty(
+                        default={}, indexed=False)))
```

Comment:
This part is not needed, the `collection_contents` will always exist as long as it is defined above (line 82).